src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * Main driver class for Duke project.
+ * Chatbot that manages a task list.
+ */
+public class Duke {
+    public static void main(String[] args) {
+        boolean isRunning = true;
+        TaskStorage taskStorage = new TaskStorage();
+        TaskList taskList;
+
+        // Load tasks from duke.TaskStorage
+        try {
+            taskList = new TaskList(taskStorage);
+        } catch (DukeException e) {
+            System.out.printf("[!] %s\n", e.getMessage());
+            return;
+        }
+
+        Ui ui = new Ui("Pong");
+        ui.init();
+
+        while (isRunning) {
+            try {
+                Parser parser = ui.getParsedInput();
+                isRunning = Duke.handleInput(parser, taskList, ui);
+            } catch (DukeException e) {
+                ui.printException(e);
+            }
+        }
+
+        ui.exit();
+    }
+
+    private static boolean handleInput(Parser parser, TaskList taskList, Ui ui) throws DukeException {
```

Comment:
I like the function in that it is succinct and it is very readable. However, I think you might want to consider refactoring the code and extracting this function into the Parser class as that class deals with making sense of the user commands.